util/processinfo.go

```diff
@@ -45,19 +45,23 @@ type ProcessInfo struct {
 // ToRowForShow returns []interface{} for the row data of "SHOW [FULL] PROCESSLIST".
 func (pi *ProcessInfo) ToRowForShow(full bool) []interface{} {
 	var info interface{}
-	if pi.Info != nil {
+	if len(pi.Info) > 0 {
 		if full {
-			info = pi.Info.(string)
+			info = pi.Info
 		} else {
-			info = fmt.Sprintf("%.100v", pi.Info.(string))
+			info = fmt.Sprintf("%.100v", pi.Info)
 		}
 	}
 	t := uint64(time.Since(pi.Time) / time.Second)
+	var db interface{}
```

Comment:
Remove the follow 4 lines, use pi.DB instead of db

Reply:
ToRowForShow here will transfer process into into row formats(interface array), nil interface is needed to represent "NULL" column value